lib/puppet/util/windows/sid.rb

```diff
@@ -75,7 +75,10 @@ def name_to_principal(name, allow_unresolved = false)
           raw_sid_bytes = sid_ptr.read_array_of_uchar(get_length_sid(sid_ptr))
         end
       rescue => e
-        Puppet.debug("Could not retrieve raw SID bytes from '#{name}': #{e.message}")
+        # Avoid debug logs pollution with valid account names
+        # https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--1300-1699-
+        # https://docs.microsoft.com/en-us/windows/win32/api/sddl/nf-sddl-convertsidtostringsidw#return-value
```

Comment:
nit: `string_to_sid_ptr` actually calls `ConvertStringSidToSidW`, the correct documentation would be https://docs.microsoft.com/en-us/windows/win32/api/sddl/nf-sddl-convertstringsidtosidw#return-value